networkx/algorithms/triads.py

```diff
@@ -0,0 +1,149 @@
+TRIAD_NAMES = ('003', '012', '102', '021D', '021U', '021C', '111D', '111U',
+               '030T', '030C', '201', '120D', '120U', '120C', '210', '300')
+
+#: The integer codes representing each type of triad.
+#:
+#: Triads that are the same up to symmetry have the same code.
+TRICODES = (1, 2, 2, 3, 2, 4, 6, 8, 2, 6, 5, 7, 3, 8, 7, 11, 2, 6, 4, 8, 5, 9,
+            9, 13, 6, 10, 9, 14, 7, 14, 12, 15, 2, 5, 6, 7, 6, 9, 10, 14, 4, 9,
+            9, 12, 8, 13, 14, 15, 3, 7, 8, 11, 7, 12, 14, 15, 8, 14, 13, 15,
+            11, 15, 15, 16)
+
+#: A dictionary mapping triad code to triad name.
+TRICODE_TO_NAME = {i: TRIAD_NAMES[code - 1] for i, code in enumerate(TRICODES)}
+
+
+def triad_graphs():
+    """Returns a dictionary mapping triad name to triad graph."""
+
+    def abc_graph():
+        """Returns a directed graph on three nodes, named ``'a'``, ``'b'``, and
+        ``'c'``.
+
+        """
+        G = nx.DiGraph()
+        G.add_nodes_from('abc')
+        return G
+
+    tg = {name: abc_graph() for name in TRIAD_NAMES}
+    tg['012'].add_edges_from([('a', 'b')])
+    tg['102'].add_edges_from([('a', 'b'), ('b', 'a')])
+    tg['102'].add_edges_from([('a', 'b'), ('b', 'a')])
+    tg['021D'].add_edges_from([('b', 'a'), ('b', 'c')])
+    tg['021U'].add_edges_from([('a', 'b'), ('c', 'b')])
+    tg['021C'].add_edges_from([('a', 'b'), ('b', 'c')])
+    tg['111D'].add_edges_from([('a', 'c'), ('c', 'a'), ('b', 'c')])
+    tg['111U'].add_edges_from([('a', 'c'), ('c', 'a'), ('c', 'b')])
+    tg['030T'].add_edges_from([('a', 'b'), ('c', 'b'), ('a', 'c')])
+    tg['030C'].add_edges_from([('b', 'a'), ('c', 'b'), ('a', 'c')])
+    tg['201'].add_edges_from([('a', 'b'), ('b', 'a'), ('a', 'c'), ('c', 'a')])
+    tg['120D'].add_edges_from([('b', 'c'), ('b', 'a'), ('a', 'c'), ('c', 'a')])
+    tg['120C'].add_edges_from([('a', 'b'), ('b', 'c'), ('a', 'c'), ('c', 'a')])
+    tg['120U'].add_edges_from([('a', 'b'), ('c', 'b'), ('a', 'c'), ('c', 'a')])
+    tg['210'].add_edges_from([('a', 'b'), ('b', 'c'), ('c', 'b'), ('a', 'c'),
+                              ('c', 'a')])
+    tg['300'].add_edges_from([('a', 'b'), ('b', 'a'), ('b', 'c'), ('c', 'b'),
+                              ('a', 'c'), ('c', 'a')])
+    return tg
+
+
+def _tricode(G, v, u, w):
+    """Returns the integer code of the given triad.
+
+    This is some fancy magic that comes from Batagelj and Mrvar's paper. It
+    treats each edge joining a pair of ``v``, ``u``, and ``w`` as a bit in
+    the binary representation of an integer.
+
+    """
+    combos = ((v, u, 1), (u, v, 2), (v, w, 4), (w, v, 8), (u, w, 16),
+              (w, u, 32))
+    return sum(x for u, v, x in combos if v in G[u])
+
+
+def triadic_census(G):
+    """Determines the triadic census of a directed graph.
+
+    The triadic census is a count of how many of the 16 possible types of
+    triads are present in a directed graph.
+
+    Parameters
+    ----------
+
+    G : digraph
+       A NetworkX DiGraph
+
+    Returns
```

Comment:
Can you remove the newlines before the first item in each section of the docstring? So these should be:

```
Parameters
----------------
G : digraph
```